runtime/lua/vim/lsp.lua

```diff
@@ -1687,7 +1689,10 @@ function lsp._vim_exit_handler()
   end
 end
 
-nvim_command('autocmd VimLeavePre * lua vim.lsp._vim_exit_handler()')
+api.nvim_create_autocmd('VimLeavePre', {
+  callback = lsp._vim_exit_handler,
```

Comment:
Callback functions cannot be inlined here because they are used in tests. I think didSave handler on `BufWritePost` could be inlined and the event could be triggered manually, but would it be safe to do with `VimLeavePre`?

Reply:
Looks like the tests are passing, so I guess it's okay.